common/clap_utils/src/preprocess.rs

```diff
@@ -0,0 +1,183 @@
+    for (key, value) in map.into_iter() {
+        let arg_key = OsString::from(format!("--{}", key));
+        if !cli_flags.contains(&arg_key) {
+            match value.as_bool() {
+                Some(true) => {
+                    args.push(arg_key);
+                }
+                Some(false) => {
+                    // no-op
+                }
+                None => {
+                    args.push(arg_key);
+                    args.push(OsString::from(f(value)?));
+                }
+            };
+        }
+    }
+    Ok(args)
+}
+
+/// A trait used to pass through the `as_bool` function for `TomlValue` and `YamlValue`.
+trait AsBool {
+    fn as_bool(&self) -> Option<bool>;
+}
+
+impl AsBool for TomlValue {
+    fn as_bool(&self) -> Option<bool> {
+        self.as_bool()
+    }
+}
+
+impl AsBool for YamlValue {
+    fn as_bool(&self) -> Option<bool> {
+        self.as_bool()
+    }
+}
+
+fn toml_value_to_string(value: TomlValue) -> Result<String, ClapPreprocessingError> {
+    let string_value = match value {
+        TomlValue::String(v) => v,
+        TomlValue::Integer(v) => v.to_string(),
+        TomlValue::Float(v) => v.to_string(),
+        TomlValue::Boolean(v) => v.to_string(),
+        TomlValue::Datetime(v) => v.to_string(),
+        TomlValue::Array(v) => v
+            .into_iter()
+            .map(toml_value_to_string)
+            .collect::<Result<Vec<_>, _>>()?
+            .join(","),
+        TomlValue::Table(_) => {
+            return Err(ClapPreprocessingError::Toml(toml::de::Error::custom(
+                "Unable to parse YAML table",
+            )))
+        }
+    };
+    Ok(string_value)
+}
+
+fn yaml_value_to_string(value: YamlValue) -> Result<String, ClapPreprocessingError> {
+    let string_value = match value {
+        YamlValue::String(v) => v,
+        YamlValue::Null => "".to_string(),
+        YamlValue::Bool(v) => v.to_string(),
+        YamlValue::Number(v) => v.to_string(),
+        YamlValue::Sequence(v) => v
+            .into_iter()
+            .map(yaml_value_to_string)
+            .collect::<Result<Vec<_>, _>>()?
+            .join(","),
+        YamlValue::Mapping(_) => {
+            return Err(ClapPreprocessingError::Yaml(serde_yaml::Error::custom(
+                "Unable to parse YAML table",
+            )))
+        }
+    };
+    Ok(string_value)
+}
+
+/// This function and `expand_args_from` are inspired by `argmatches::expand_args_from` and
+/// `argmatches::expand_args_from` respectively, but differ in the following ways:
+///
+/// - This has no recursion.
+/// - `parser` must return a `Result<Vec<OsString>, ClapPreprocessingError>` as opposed to a `Vec<Argument>`.
+/// - This tracks and passes along to `parser` all cli flags provided by the user.
+/// - This searches for config files based on a provided flag name rather than a prefix.
+pub fn expand_args<F>(
+    parser: F,
+    flag_name: &str,
+) -> Result<Vec<std::ffi::OsString>, ClapPreprocessingError>
```

Comment:
`OsString` is also imported so we can delete the path prefix (there's a few of these in this file)

```suggestion
) -> Result<Vec<OsString>, ClapPreprocessingError>
```